src/system/SystemPacketBuffer.cpp

```diff
@@ -652,14 +658,28 @@ PacketBufferHandle PacketBufferHandle::NewWithData(const void * aData, size_t aD
     PacketBufferHandle buffer = New(aDataSize + aAdditionalSize, aReservedSize);
     if (buffer.mBuffer != nullptr)
     {
-        memcpy(buffer.mBuffer->payload, aData, aDataSize);
 #if CHIP_SYSTEM_CONFIG_USE_LWIP
         // Checks in the New() call catch buffer allocations greater
         // than UINT16_MAX for LwIP based platforms.
-        buffer.mBuffer->len = buffer.mBuffer->tot_len = static_cast<uint16_t>(aDataSize);
+        buffer.mBuffer->tot_len = static_cast<uint16_t>(aDataSize);
+#else
+        buffer.mBuffer->tot_len = aDataSize;
+#endif
+        PacketBuffer * currentBuffer = buffer.mBuffer;
```

Comment:
Please give a detailed comment above this logic explaining the copying of the pbuf chain for easy future context.

Reply:
Done.